### PR TITLE
Add new InfluxDB chart

### PR DESCRIPTION
**Pull request by juan131**
This PR adds a new Helm chart to the catalog: InfluxDB

#### What is InfluxDB?

InfluxDB is an open source time-series database designed to handle large write and read loads in real-time.

#### Special Notes for Reviewers

This chart support two different architecture setups: **standalone** and **high-availability**. please put extra emphasis on this peculiarity.

### CODE REVIEW

File: bitnami/influxdb/README.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+| `influxdb.securityContext.enabled`          | Enable security context for InfluxDB                                                                                                                                                      | `true`                                                  |
+| `influxdb.securityContext.fsGroup`          | Group ID for the InfluxDB filesystem                                                                                                                                                      | `1001`                                                  |
+| `influxdb.securityContext.runAsUser`        | User ID for the InfluxDB container                                                                                                                                                        | `1001`                                                  |
+| `influxdb.resources`                        | The [resources] to allocate for container                                                                                                                                                 | `{}`                                                    |
```
Comment (javsalgar) [inline bitnami/influxdb/README.md]: Are we missing a link in [resources]?

File: bitnami/influxdb/README.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+The allowed extensions are `.sh`, and `.txt`.
+
+### Persistence
+
+The data is persisted by default using PVC(s). You can disable the persistence setting the `persistence.enabled` parameter to `false`.
+A default `StorageClass` is needed in the Kubernetes cluster to dynamically provision the volumes. Specify another StorageClass in the `persistence.storageClass` or set `persistence.existingClaim` if you have already existing persistent volumes to use.
+
+## Upgrade
+
+It's necessary to specify the existing passwords while performing a upgrade to ensure the secrets are not updated with invalid randomly generated passwords. Remember to specify the existing values of the `adminUser.pwd`, `user.pwd`, `readUser.pwd` and `writeUser.pwd` parameters when upgrading the chart:
```
Comment (javsalgar) [inline bitnami/influxdb/README.md]: an upgrade

File: bitnami/influxdb/README.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+| `persistence.accessMode`                    | Access mode of data volume                                                                                                                                                                | `ReadWriteOnce`                                         |
+| `persistence.size`                          | Size of data volume                                                                                                                                                                       | `8Gi`                                                   |
+| `volumePermissions.enabled`                 | Enable init container that changes the owner and group of the persistent volume mountpoint to `runAsUser:fsGroup`                                                                         | `false`                                                 |
+| `volumePermissions.image.registry`          | Init container volume-permissions image registry                                                                                                                                          | `docker.io`                                             |
```
Comment (javsalgar) [inline bitnami/influxdb/README.md]: should volume-permission have backticks? `volume-permissions`

File: bitnami/influxdb/templates/influxdb/standalone-dpl.yaml [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+  selector:
+    matchLabels: {{ include "influxdb.matchLabels" . | nindent 6 }}
+      app.kubernetes.io/component: influxdb
+  template:
+    metadata:
+      labels: {{ include "influxdb.labels" . | nindent 8 }}
+        app.kubernetes.io/component: influxdb
+    spec:
+{{- include "influxdb.imagePullSecrets" . | indent 6 }}
+      affinity:
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/influxdb/templates/influxdb/standalone-dpl.yaml]: I would by the "tpl + toYaml" approach **only if** by default we're including in the **values.yaml** the equivalent values so when you install the chart (without modifying any value) the result is:
```
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - weight: 1
            podAffinityTerm:
              topologyKey: kubernetes.io/hostname
              labelSelector:
                matchLabels:
                  app.kubernetes.io/name: influxdb
                  app.kubernetes.io/instance: my-release
                  app.kubernetes.io/component: XXX
```
I mean, by default the "affinity.podAntiAffinity" should be defined.

File: bitnami/influxdb/templates/influxdb/standalone-dpl.yaml [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        {{- end }}
+        env:
+        - name: BITNAMI_DEBUG
+          value: {{ ternary "true" "false" .Values.image.debug | quote }}
+        - name: INFLUXDB_HTTP_AUTH_ENABLED
+          value: {{ .Values.authEnabled | quote }}
+        {{- if .Values.adminUser.user }}
+        - name: INFLUXDB_ADMIN_USER
+          value: {{ .Values.adminUser.user | quote }}
+        {{- if .Values.adminUser.usePasswordFile }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/influxdb/templates/influxdb/standalone-dpl.yaml]: +1

File: bitnami/influxdb/values.yaml [hunk 6]
```diff
@@ -0,0 +1,475 @@
...
+  ##
+  securityContext:
+    enabled: true
+    fsGroup: 1001
+    runAsUser: 1001
+
+  ## InfluxDB Relay pods' resource requests and limits
+  ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
+  ##
+  resources: {}
```
Comment (javsalgar) [inline bitnami/influxdb/values.yaml]: Change resources so it looks like this

```
  resources:
    # We usually recommend not to specify default resources and to leave this as a conscious
    # choice for the user. This also increases chances charts run on environments with little
    # resources, such as Minikube. If you do want to specify resources, uncomment the following
    # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
    limits: {}
    #   cpu: 100m
    #   memory: 128Mi
    requests: {}
    #   cpu: 100m
    #   memory: 128Mi
```

This avoids this linter issue https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.comments_indentation

File: bitnami/influxdb/templates/influxdb/statefulset.yaml [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+kind: StatefulSet
+metadata:
+  name: {{ include "influxdb.fullname" . }}
+  labels: {{ include "influxdb.labels" . | nindent 4 }}
+    app.kubernetes.io/component: influxdb
+spec:
+  serviceName: {{ include "influxdb.fullname" . }}-headless
+  replicas: {{ .Values.influxdb.replicaCount }}
+  updateStrategy:
+    type: {{ .Values.influxdb.updateStrategy }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/influxdb/templates/influxdb/statefulset.yaml]: I think that's used on deployments but most of our **statefulsets** follow this approach.

File: bitnami/influxdb/templates/influxdb/statefulset.yaml [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+          successThreshold: {{ .Values.influxdb.readinessProbe.successThreshold }}
+          failureThreshold: {{ .Values.influxdb.readinessProbe.failureThreshold }}
+        {{- end }}
+        {{- if .Values.influxdb.resources }}
+        resources: {{- toYaml .Values.influxdb.resources | nindent 10 }}
+        {{- end }}
+        volumeMounts:
+        {{- if or (.Files.Glob "files/conf/influxdb.conf") .Values.influxdb.configuration .Values.influxdb.existingConfiguration  }}
+        - name: influxdb-config
+          mountPath: /opt/bitnami/influxdb/etc
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/influxdb/templates/influxdb/statefulset.yaml]: Yes, only one configuration file

File: bitnami/influxdb/templates/influxdb/statefulset.yaml [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+          {{- if .Values.readUser.usePasswordFile }}
+          - key: read-user-password
+            path: read-user-password
+          {{- end }}
+          {{- if .Values.writeUser.usePasswordFile }}
+          - key: write-user-password
+            path: write-user-password
+          {{- end }}
+      {{- end }}
+  {{- if and .Values.persistence.enabled .Values.persistence.existingClaim }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/influxdb/templates/influxdb/statefulset.yaml]: We don't want that. Each InfluxDB replicas is treated as an independent database.

File: bitnami/influxdb/templates/influxdb/secrets.yaml [hunk 10]
```diff
@@ -0,0 +1,19 @@
+{{- if (not .Values.existingSecret) -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "influxdb.fullname" . }}
+  labels: {{ include "influxdb.labels" . | nindent 4 }}
+type: Opaque
+data:
+  admin-user-password: {{ ternary (randAlphaNum 10) .Values.adminUser.pwd (empty .Values.adminUser.pwd) | b64enc | quote }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/influxdb/templates/influxdb/secrets.yaml]: I can add support for the `forcePassword` parameter if you want

File: bitnami/influxdb/README.md [hunk 11]
```diff
@@ -0,0 +1,348 @@
...
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release. Use the option `--purge` to delete all history too.
+
+## Standalone vs High Availability architecture
+
+You can install the InfluxDB chart with two different architecture setups: `standalone` or `high-availability`
+
+```console
+$ helm install --name my-release bitnami/minio --set architecture="standalone"
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/influxdb/README.md]: Leaving it for the moment

File: bitnami/influxdb/README.md (hunk 3 above)
Comment (juan131) [reply bitnami/influxdb/README.md]: Why? We're using camelCase everywhere

Comment (juan131) [reply bitnami/influxdb/README.md]: Why?

File: bitnami/influxdb/README.md (hunk 1 above)
Comment (juan131) [reply bitnami/influxdb/README.md]: Nope, I just copied it from other chart that uses that.

File: bitnami/influxdb/templates/NOTES.txt [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+{{- end }}
+
+To connect to your database run the following commands:
+
+    {{- if eq .Values.architecture "high-availability" }}
+
+    (write operations):
+
+    kubectl run {{ include "influxdb.fullname" . }}-client --rm --tty -i --restart='Never' --namespace {{ .Release.Namespace }} {{ if .Values.authEnabled }}--env="INFLUX_USERNAME={{ .Values.adminUser.user }}" --env="INFLUX_PASSWORD=$ADMIN_PASSWORD"{{ end }} \
+        {{ if and (.Values.networkPolicy.enabled) (not .Values.networkPolicy.allowExternal) }}--labels="{{ include "influxdb.fullname" . }}-client=true" {{ end }}--image bitnami/influxdb:1 \
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/influxdb/templates/NOTES.txt]: Do we have a way to automate updating this?

File: bitnami/influxdb/files/conf/README.md [hunk 13]
```diff
@@ -0,0 +1,6 @@
+Place your InfluxDB and InfluxDB Relay configuration files here. These will not be used in case the values *existingConfiguration*, *relay.existingConfiguration* are used.
+
+More information can be found in the links below:
+
+- [InfluxDB Configuration File](https://github.com/bitnami/bitnami-docker-influxdb#configuration-file)
+- [InfluxDB Relay Configuration File](https://github.com/bitnami/bitnami-docker-influxdb-relay#configuration-file)
```
Comment (alemorcuq) [inline bitnami/influxdb/files/conf/README.md]: This link is dead.

File: bitnami/influxdb/templates/NOTES.txt [hunk 14]
```diff
@@ -0,0 +1,144 @@
+
+** Please be patient while the chart is being deployed **
+
+{{- if eq .Values.architecture "high-availability" }}
+
+InfluxDB can be accessed through following DNS names from within your cluster:
+
+    InfluxDB Relay (write operations): {{ include "influxdb.fullname" . }}-relay.{{ .Release.Namespace }}.svc.cluster.local (port {{ .Values.relay.service.port }})
+    InfluxDB servers (read operations):  {{ include "influxdb.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local (port {{ .Values.influxdb.service.port }})
```
Comment (alemorcuq) [inline bitnami/influxdb/templates/NOTES.txt]: I think it's worth making `cluster.local` a parameter, such as `.Values.clusterDomain`.

File: bitnami/influxdb/templates/NOTES.txt (hunk 12 above)
[the remark this answers is not in the document]
Comment (alemorcuq) [reply bitnami/influxdb/templates/NOTES.txt]: Maybe `.Values.image.tag`?

File: bitnami/influxdb/templates/influxdb/configmap.yaml [hunk 15]
```diff
@@ -0,0 +1,16 @@
+{{- if and (or (.Files.Glob "files/conf/influxdb.conf") .Values.influxdb.configuration) (not .Values.influxdb.existingConfiguration) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "influxdb.fullname" . }}
+  labels: {{ include "influxdb.labels" . | nindent 4 }}
+    app.kubernetes.io/component: influxdb
+data:
+{{- if (.Files.Glob "files/conf/influxdb.conf") }}
+{{ (.Files.Glob "iles/conf/influxdb.conf").AsConfig | indent 2 }}
```
Comment (alemorcuq) [inline bitnami/influxdb/templates/influxdb/configmap.yaml]: Typo: `files/conf...`

File: bitnami/influxdb/templates/influxdb/configmap.yaml [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+{{- else if .Values.influxdb.configuratio }}
```
Comment (alemorcuq) [inline bitnami/influxdb/templates/influxdb/configmap.yaml]: Typo: `configuration`

File: bitnami/influxdb/templates/influxdb/standalone-dpl.yaml [hunk 17]
```diff
@@ -0,0 +1,253 @@
...
+        - name: influxdb
+          image: {{ include "influxdb.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          {{- if .Values.influxdb.securityContext.enabled }}
+          securityContext:
+            runAsUser: {{ .Values.influxdb.securityContext.runAsUser }}
+          {{- end }}
+          env:
+            - name: BITNAMI_DEBUG
+              value: {{ ternary "true" "false" .Values.image.debug | quote }}
```
Comment (alemorcuq) [inline bitnami/influxdb/templates/influxdb/standalone-dpl.yaml]: Why is this a ternary and not just `.Values.image.debug`? Doesn't seem consistent with other values. E.g.

```yaml
- name: INFLUXDB_HTTP_AUTH_ENABLED
  value: {{ .Values.authEnabled | quote }}
```

File: bitnami/influxdb/templates/influxdb/standalone-dpl.yaml [hunk 18]
```diff
@@ -0,0 +1,253 @@
...
+          livenessProbe:
+            exec:
+              command:
+                - bash
+                - -c
+                - |
+                  if [[ -f "${INFLUXDB_ADMIN_USER_PASSWORD_FILE:-}" ]]; then
+                      export INFLUXDB_ADMIN_USER_PASSWORD="$(< "${INFLUXDB_ADMIN_USER_PASSWORD_FILE}")"
+                  fi
+                  {{ if .Values.authEnabled }}INFLUX_USERNAME="$INFLUXDB_ADMIN_USER" INFLUX_PASSWORD="$INFLUXDB_ADMIN_USER_PASSWORD"{{ end }} influx -host 127.0.0.1 -port 8086 -execute "SHOW DATABASES"
```
Comment (alemorcuq) [inline bitnami/influxdb/templates/influxdb/standalone-dpl.yaml]: Just thinking out loud: does it make sense to make the probes commands also customizable in the `values.yaml`?

File: bitnami/influxdb/templates/influxdb/statefulset.yaml [hunk 19]
```diff
@@ -0,0 +1,273 @@
...
+metadata:
+  name: {{ include "influxdb.fullname" . }}
+  labels: {{- include "influxdb.labels" . | nindent 4 }}
+    app.kubernetes.io/component: influxdb
+spec:
+  serviceName: {{ include "influxdb.fullname" . }}-headless
+  replicas: {{ .Values.influxdb.replicaCount }}
+  updateStrategy:
+    type: {{ .Values.influxdb.updateStrategy }}
+    {{- if (eq "Recreate" .Values.influxdb.updateStrategy) }}
```
Comment (alemorcuq) [inline bitnami/influxdb/templates/influxdb/statefulset.yaml]: I don't think `Recreate` is a valid `updateStrategy` for `StatefulSets`: https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#updating-statefulsets

File: bitnami/influxdb/templates/relay/deployment.yaml [hunk 20]
```diff
@@ -0,0 +1,97 @@
+{{- if eq .Values.architecture "high-availability" }}
```
Comment (alemorcuq) [inline bitnami/influxdb/templates/relay/deployment.yaml]: Nice. I didn't know you could put the templates into folders! Much better than doing `relay-deployment.yaml` or `relay-svc.yaml`.

File: bitnami/influxdb/values.yaml [hunk 21]
```diff
@@ -0,0 +1,490 @@
...
+##
+volumePermissions:
+  enabled: false
+  ## Bitnami Minideb image
+  ## ref: https://hub.docker.com/r/bitnami/minideb/tags/
+  ##
+  image:
+    registry: docker.io
+    repository: bitnami/minideb
+    tag: latest
```
Comment (alemorcuq) [inline bitnami/influxdb/values.yaml]: We don't use the `latest` tag. Use `stretch` instead.

File: bitnami/influxdb/README.md [hunk 22]
```diff
@@ -0,0 +1,352 @@
...
+          ▼                            ▼
+      ┌────────────────┐  ┌────────────────┐
+      │ InfluxDB Relay │  │    InfluxDB    │
+      │      svc       │  │      svc       │
+      └───────┬─┬──────┘  └─────┬─┬────────┘
+      ┌────── │─|───────────────|─│───────┐
+      |       │ |               | │       ▼
+┌─────┴────┐  │ |               | │  ┌──────────┐
+│ InfluxDB │  │ |               | │  │ InfluxDB │
+│  Relay   │◀─┘ |               | └─▶│  Server  │
```
Comment (alemorcuq) [inline bitnami/influxdb/README.md]: This is great, by the way!!

File: bitnami/influxdb/templates/NOTES.txt (hunk 14 above)
Comment (juan131) [reply bitnami/influxdb/templates/NOTES.txt]: I don't see any benefit on doing so.

File: bitnami/influxdb/templates/influxdb/standalone-dpl.yaml (hunk 17 above)
Comment (juan131) [reply bitnami/influxdb/templates/influxdb/standalone-dpl.yaml]: This "ternary" way also covers the case where `.Values.image.debug` is not defined

File: bitnami/influxdb/templates/influxdb/standalone-dpl.yaml (hunk 18 above)
Comment (juan131) [reply bitnami/influxdb/templates/influxdb/standalone-dpl.yaml]: We can customize everything in the yaml, even if we use "exec" or "httpGet" probes but the question is: Are we adding **real** value in it? I don't think so

File: bitnami/influxdb/templates/NOTES.txt (hunk 14 above)
Comment (alemorcuq) [reply bitnami/influxdb/templates/NOTES.txt]: We have had support tickets in the past due to hardcoding `cluster.local`. Here's an example: https://github.com/helm/charts/issues/13230#issuecomment-493455435

**Pull request merged**